<a href="https://colab.research.google.com/github/nedburke47/PracticingForClass/blob/master/Project_Part_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [28]:
%load_ext sql
%sql sqlite:///deliveryService.db

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


'Connected: @deliveryService.db'

In [29]:
%%sql  
DROP TABLE IF EXISTS Cooks;
create table Cooks (CookSSN PRIMARY KEY, First Name text, Cook Last Name text, DOB text, Position text);

DROP table IF EXISTS Orders;
create table Orders (OrderID text PRIMARY KEY, Dish text, Price real, Cook Last Name text, CustomerService Last Name text, Driver Last Name text);

DROP table IF EXISTS Customers;
create table Customers (IDNumber text PRIMARY KEY, First Name text, Last Name text, DOB text, OrderID text);

DROP table IF EXISTS Drivers;
create table Drivers ( DriverSSN text PRIMARY KEY, First Name text, Last Name text, License number text, DOB text);

DROP table IF EXISTS ElectricCars;
create table ElectricCars (License Plate Numbers text PRIMARY KEY, Driver Last Name text, Deliveries text, Model text, Color text);

DROP table IF EXISTS CustomerService;
create table CustomerService (CustomerServiceSSN text PRIMARY KEY, First Name text, Last Name text, DOB text, Position text );








 * sqlite:///deliveryService.db
Done.
Done.
Done.
Done.
Done.
Done.
Done.
Done.
Done.
Done.
Done.
Done.


[]

In [30]:
%%sql
insert into Customers values ('19902', 'James', 'Smith', '07/06/2000' , '12910');
insert into Customers values ('19203', 'Sarah', 'Williams', '08/19/1992' , '10920');
insert into Customers values ('13180', 'Ed', 'Swayze', '02/20/1994' , '12893');
insert into Customers values ('18920', 'Maggie', 'Jackman', '04/22/1981' , '10279');
insert into Customers values ('18302', 'Patrick', 'Jackson', '04/22/1981' , '19279');



insert into Cooks values ('201-191-1819', 'John', 'Prescott', '7/11/1991', 'Line Cook');
insert into Cooks values ('102-111-1940', 'Jessica', 'Taylor' , '01/14/1983', 'Line Cook');
insert into Cooks values ('782-105-1840', 'Jamie', 'Burke' , '11/07/1999', 'Line Cook');
insert into Cooks values ('132-121-5710', 'Elysia', 'McClane' , '02/29/1997', 'Line Cook');
insert into Cooks values ('122-951-3820', 'Anthony', 'Vallor' , '04/14/1988', 'Line Cook');

insert into Orders values('12910', 'Steak', '23.99', 'Prescott', 'Fitz', 'Franklin');
insert into Orders values('10920', 'Fish', '17.99', 'Taylor', 'West', 'Montana');
insert into Orders values('10279', 'Soup', '7.99', 'Prescott', 'West', 'Montana');
insert into Orders values('12893', 'Steak', '23.99', 'Prescott', 'Fitz', 'Franklin');
insert into Orders values('12894', 'Cheesecake', '8.99', 'Taylor', 'West', 'Franklin');


insert into Drivers values ('211781010', 'John', 'Franklin', '04/22/2001' , '11084027');
insert into Drivers values ('182092733', 'Lucy', 'Montana', '08/21/1999' , '12850378');
insert into Drivers values ('282017301', 'Antoine', 'Jay', '08/01/1991' , '14871202');
insert into Drivers values ('128930444', 'Thomas', 'Alvarez', '05/22/1989' , '28103820');
insert into Drivers values ('123939101', 'Michael', 'Morrow', '04/24/1971' , '28103830');

insert into ElectricCars values ('kit-1830', 'Montana', '2', 'Tesla Model X' , 'Blue');
insert into ElectricCars values ('ghe-3791', 'Franklin', '3', 'Tesla Model S' , 'Grey');
insert into ElectricCars values ('ahd-1839', 'Jay', '0', 'Nissan Leaf' , 'Green');
insert into ElectricCars values ('ftj-8966', 'Alvarez', '0', 'Chevy Volt' , 'Black');
insert into ElectricCars values ('fvu-9689', 'Morrow', '0', 'Tesla Model X' , 'Grey');

insert into CustomerService values( '121-01-2910' , 'Mike', 'West', '02/17/1975', 'Represenative');
insert into CustomerService values('192-19-2840', 'Craig', 'Fitz' , '07/8/1995', 'Represenative');
insert into CustomerService values('281-20-1830', 'Connor', 'Wells' , '01/9/1992', 'Represenative');
insert into CustomerService values('582-29-2330', 'Angie', 'James' , '09/13/1985', 'Represenative');
insert into CustomerService values('129-59-2280', 'John', 'Trot' , '7/30/1982', 'Represenative');

 * sqlite:///deliveryService.db
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]


![ER Diagram](https://raw.githubusercontent.com/nedburke47/PracticingForClass/master/Yet%20Another%20Delivery%20Project.png)

In [31]:
%%sql
select AVG(Price) from Orders

 * sqlite:///deliveryService.db
Done.


[(16.589999999999996,)]

“What is the average cost of all the orders”, we wanted to get the average cost of the orders. It would be important because we need to know how money is coming in for orders and how much revenue is getting pulled in for the restaurant. The average price of the orders is $16.59. The results matched our experience because the calculations add up. 

In [32]:
%%sql
select Driver, COUNT(Driver) from Orders GROUP by Driver
 


 * sqlite:///deliveryService.db
Done.


[('Franklin', 3), ('Montana', 2)]


“How many orders have the drivers delivered?”, we found out how many orders each driver delivered of the drivers that were on duty that day. Franklin had 3 deliveries while Montana had 2.  Business-wise, this is very important to know how many deliveries are being made because it tells us how efficient our drivers are. This question helped with finding the efficiency of our drivers. These results do match our experience because it correctly calculated the proper drivers. 


In [33]:
%%sql
select SUM(Price) from Orders

 * sqlite:///deliveryService.db
Done.


[(82.94999999999999,)]

"What is the total sum of the orders?", The total sum of the prices of all the food we cold is important because it shows out total profits for the day. This is useful, because we can compare profits from day to day and use it to figure out which days we are busiest and which days we are not busy, which will allow us to schedule staff accordingly. The results matched our experience because the calculations add up. 